docs/design/seccomp.md

```diff
@@ -0,0 +1,380 @@
+	Value    int64           `json:"value"`
+	ValueTwo int64           `json:"valueTwo"`
+	Op       SeccompOperator `json:"op"`
+}
+```
+
+Along with this API change, there will be supporting code changes:
+
+1.  New REST API implementation
+2.  Client code
+2.  Validations/conversions
+3.  `kubectl` support (including describer)
+
+### Pod API changes
+
+The container `securityContext` must be changed to allow specifying a seccomp profile:
+
+```go
+// SecurityContext holds security configuration that will be applied to a container.
+// Some fields are present in both SecurityContext and PodSecurityContext.  When both
+// are set, the values in SecurityContext take precedence.
+type SecurityContext struct {
+  // other fields omitted
+  // SeccompProfileName is the name of the seccomp profile this pod should run under.
+  SeccompProfileName string `json:"seccompProfileName,omitempty"`
+}
+```
+
+The kubelet docker runtime implementation in `pkg/kubelet/dockertools` will
+require code changes to implement setting the seccomp profile when a container
+is run.  This API change should not affect the kubelet itself.
+
+The kubectl describer for pods should change to include the new field.
+
+### Pod Security Policy changes
+
+The `PodSecurityPolicy` type should be changed to control seccomp profiles:
+
+```go
+// PodSecurityPolicySpec defines the policy enforced.
+type PodSecurityPolicySpec struct {
+  // other fields omitted
+  Seccomp *SeccompStrategyOptions `json:"seccomp,omitempty"`
+}
+
+// Defines the seccomp strategy type and any options used to create the strategy.
+type SeccompStrategyOptions struct {
+	// Rule is the strategy that will dictate the allowable seccomp profile name values that may be set.
+	Rule SeccompStrategy `json:"rule"`
+	// Profiles are a list of profile names that may be used.
+	Profiles []string `json:"profiles,omitempty"`
+}
+
+// SeccompStrategy denotes strategy types for generating SeccompProfileName values for a
+// SecurityContext.
+type SeccompStrategy string
+
+const (
+	// container must run as one of a set of profile names.
+	SeccompStrategyMustRunAs SeccompStrategy = "MustRunAs"
+	// container may make requests for any profile name.
+	SeccompStrategyRunAsAny SeccompStrategy = "RunAsAny"
+)
+```
+
+## Examples
+
+### Use in a pod
+
+Here's an example of a pod that uses a profile called `example-explorer-profile`:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: explorer
+spec:
+  containers:
+    - name: explorer
+      image: gcr.io/google_containers/explorer:1.0
+      args: ["-port=8080"]
+      ports:
+        - containerPort: 8080
+          protocol: TCP
+      securityContext:
+        seccompProfileName: example-explorer-profile
+      volumeMounts:
+        - mountPath: "/mount/test-volume"
+          name: test-volume
+  volumes:
+    - name: test-volume
+      emptyDir: {}
+```
+
+### Unconfined profile
+
+An unconfined profile can be implemented with a single default action:
```

Comment:
is this necessary? in docker we don't apply a profile when unconfined is denoted which is kinda the same thing